samples/src/main/kotlin/com/example/sample/KotlinActivity.kt

```diff
@@ -132,9 +147,36 @@ class KotlinActivity : AppCompatActivity() {
         dataBrokerConnection?.subscribe(property, propertyListener)
     }
 
-    fun disconnect() {
-        dataBrokerConnection?.disconnectListeners?.unregister(disconnectListener)
-        dataBrokerConnection?.disconnect()
-        dataBrokerConnection = null
+    // region: Specifications
+    fun fetchSpecification() {
+        lifecycleScope.launch {
+            try {
+                val vssSpeed = VssVehicle.VssSpeed()
+                val updatedSpeed = dataBrokerConnection?.fetch(vssSpeed)
+                val speed = updatedSpeed?.value
+            } catch (e: DataBrokerException) {
+                // handle errors
+            }
+        }
+    }
+
+    fun updateSpecification() {
+        lifecycleScope.launch {
+            val vssSpeed = VssVehicle.VssSpeed(value = 100f)
+            val updatedSpeed = dataBrokerConnection?.update(vssSpeed)
+        }
+    }
+
+    fun subscribeSpecification() {
+        val vssSpeed = VssVehicle.VssSpeed(value = 100f)
+        dataBrokerConnection?.subscribe(
+            vssSpeed,
+            listener = object : VssSpecificationListener<VssVehicle.VssSpeed> {
+                override fun onSpecificationChanged(vssSpecification: VssVehicle.VssSpeed) {
+                    val speed = vssSpeed.value
+                }
+            },
+        )
     }
+    // endregion
```

Comment:
Not totally happy to mix both approaches here, but it probably is better than adding redundant code for both approaches separately. 

Anyway when you add something to KotlinActivity you also need to make sure to add it into the corresponding JavaActivity

Reply:
Added the java samples. Also changed some API to Collection instead of list. Makes it easier to use the Java API with  `Collections.singleton(Types.Field.FIELD_VALUE)`